src/ddscxx/src/org/eclipse/cyclonedds/core/cdr/cdr_stream.cpp

```diff
@@ -144,6 +133,45 @@ bool cdr_stream::finish_member(const entity_properties_t &props, member_id_set &
   return true;
 }
 
+void cdr_stream::set_mode(stream_mode mode, key_mode key)
+{
+  assert(key != key_mode::unset);
+  m_mode = mode;
+  m_key = key;
+  reset();
+}
+
+size_t cdr_stream::incr_position(size_t incr_by)
```

Comment:
I image this function (and perhaps some of the others that were moved here) gets called very often and it is still very simple. I would expect inlining this one to have performance benefit.

(Also: `align` which definitely gets called often. That one is fairly complex, but specializing it for read vs write and doing the complex bits in a separate function might be interesting.)

Reply:
Inlining some of these functions will probably improve performance, though to what degree, I am not certain.

Compilers are very smart these days, and I suspect that especially for small functions, the inlining is being done implicitly for any significant optimization.